samples/nrf9160/gnss/src/assistance.c

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <zephyr.h>
+#include <modem/modem_info.h>
+#include <modem/modem_jwt.h>
+#include <net/nrf_cloud_rest.h>
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+#include <net/nrf_cloud_agps.h>
+#endif /* CONFIG_NRF_CLOUD_AGPS */
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <pm_config.h>
+#include <date_time.h>
+#include <net/nrf_cloud_pgps.h>
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+#include "assistance.h"
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+static char agps_data_buf[3500];
+#endif
+static char jwt_buf[600];
+static char rx_buf[2048];
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+static struct nrf_modem_gnss_agps_data_frame agps_need;
+static struct gps_pgps_request pgps_request;
+static struct nrf_cloud_pgps_prediction *prediction;
+static struct k_work get_pgps_data_work;
+static struct k_work inject_pgps_data_work;
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+static struct k_work_q *work_q;
+static bool assistance_active;
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+static int serving_cell_info_get(struct lte_lc_cell *serving_cell)
+{
+	int err;
+
+	err = modem_info_init();
+	if (err) {
+		return err;
+	}
+
+	char resp_buf_cellid[MODEM_INFO_MAX_RESPONSE_SIZE];
+	char resp_buf_tac[MODEM_INFO_MAX_RESPONSE_SIZE];
+	char resp_buf_cops[MODEM_INFO_MAX_RESPONSE_SIZE];
```

Comment:
Same comment here that popped up with Location lib. You can save mem by using only one buffer and parsing the data as you go.

Reply:
Will do.